chatroom-socket-server/server.js

```diff
@@ -0,0 +1,43 @@
+const port = 8000;
+const httpServer = require("http").createServer((req, res) => {}).listen(port);
+const io = require("socket.io")(httpServer, {
+  cors: {
+    origin: "http://localhost:3000",
+  },
+});
+
+// when socket.connect on react gets called, this runs
+io.use((socket, next) => {
+    console.log("connected " + socket.handshake.auth.username)
+    const username = socket.handshake.auth.username
+    if (!username) {
+        return next(new Error("invalid"))
+    }
+    socket.username = username;
+    console.log("connect is called and received by server")
```

Comment:
Possibly add for logging purposes of the chat connections the username when saying when it connects?